server/src/test/java/io/crate/expression/scalar/ArrayMinFunctionTest.java

```diff
@@ -0,0 +1,59 @@
+package io.crate.expression.scalar;
+
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormat;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.crate.testing.Asserts.assertThrows;
+
+public class ArrayMinFunctionTest extends AbstractScalarFunctionsTest {
+
+    @Test
+    public void test_array_returns_min_element() {
+        for(DataType t: DataTypes.PRIMITIVE_TYPES) {
+
+            var expression = new StringBuilder("array_min(cast([")
+                .append(listToCommaSeparatedString(List.of(3,2,1), t))
```

Comment:
I wonder if we should instead extend `assertEvaluate` to also support parameter placeholders, so that you could do something like `assertEvaluate("array_min(?)", expected, Literal.of([1, 2, 3]))`

It already supports column references, but it only works for the columns that are defined in the base class (see `prepareFunctions` in `AbstractScalarFunctionsTest`)

Reply:
Done (exclusive implementation, Literals are either params or refs, if we want both then probably need another overload 
with 2 Lists -one for params, one for refs, not sure we have such test case).

Facing casting exceptions, something to do with module.register. Converted PR to draft for now

Reply:
Done